compatibility/bazel_tools/daml_ledger/Main.hs

```diff
@@ -0,0 +1,256 @@
+import Test.Tasty.Options (IsOption(..), OptionDescription(..), mkOptionCLParser)
+import Test.Tasty.HUnit (testCaseSteps)
+import qualified Bazel.Runfiles
+import qualified Data.Aeson as Aeson
+import qualified Data.List as List
+import qualified Data.Map as Map
+import qualified Data.Text as T
+import qualified Web.JWT as JWT
+
+import Sandbox
+
+data Tools = Tools
+  { daml :: FilePath
+  , sandboxConfig :: SandboxConfig
+  }
+
+newtype DamlOption = DamlOption FilePath
+instance IsOption DamlOption where
+  defaultValue = DamlOption $ "daml"
+  parseValue = Just . DamlOption
+  optionName = Tagged "daml"
+  optionHelp = Tagged "runfiles path to the daml executable"
+
+newtype SandboxOption = SandboxOption FilePath
+instance IsOption SandboxOption where
+  defaultValue = SandboxOption $ "sandbox"
+  parseValue = Just . SandboxOption
+  optionName = Tagged "sandbox"
+  optionHelp = Tagged "runfiles path to the sandbox executable"
+
+newtype SandboxArgsOption = SandboxArgsOption { unSandboxArgsOption :: [String] }
+instance IsOption SandboxArgsOption where
+  defaultValue = SandboxArgsOption []
+  parseValue = Just . SandboxArgsOption . (:[])
+  optionName = Tagged "sandbox-arg"
+  optionHelp = Tagged "extra arguments to pass to sandbox executable"
+  optionCLParser = concatMany (mkOptionCLParser mempty)
+    where concatMany = fmap (SandboxArgsOption . concat) . many . fmap unSandboxArgsOption
+
+newtype CertificatesOption = CertificatesOption FilePath
+instance IsOption CertificatesOption where
+  defaultValue = CertificatesOption $ "certificates"
+  parseValue = Just . CertificatesOption
+  optionName = Tagged "certs"
+  optionHelp = Tagged "runfiles path to the certificates directory"
+
+withTools :: (IO Tools -> TestTree) -> TestTree
+withTools tests = do
+  askOption $ \(DamlOption damlPath) -> do
+  askOption $ \(SandboxOption sandboxPath) -> do
+  askOption $ \(SandboxArgsOption sandboxArgs) -> do
+  askOption $ \(CertificatesOption certificatesPath) -> do
+  let createRunfiles :: IO (FilePath -> FilePath)
+      createRunfiles = do
+        runfiles <- Bazel.Runfiles.create
+        mainWorkspace <- fromMaybe "compatibility" <$> lookupEnv "TEST_WORKSPACE"
+        pure (\path -> Bazel.Runfiles.rlocation runfiles $ mainWorkspace </> path)
+  withResource createRunfiles (\_ -> pure ()) $ \locateRunfiles -> do
+  let tools = do
+        daml <- locateRunfiles <*> pure damlPath
+        sandboxBinary <- locateRunfiles <*> pure sandboxPath
+        sandboxCertificates <- locateRunfiles <*> pure certificatesPath
+        let sandboxConfig = defaultSandboxConf
+              { sandboxBinary
+              , sandboxArgs
+              , sandboxCertificates
+              }
+        pure Tools
+          { daml
+          , sandboxConfig
+          }
+  tests tools
+
+newtype SdkVersion = SdkVersion String
+instance IsOption SdkVersion where
+  defaultValue = SdkVersion "1.0.0"
```

Comment:
Do we ever want to use the default value here? If not, would it make sense to set this to `error` to make sure it blows up if we do not set it?

Reply:
Good point. `tasty` doesn't allow options without default values, but `error` is a good enough work around for a test runner.